### PR TITLE
Add example token contract to the parse test

### PR DESCRIPTION
**Pull request by thealmarty**

### CODE REVIEW

**Comment by thealmarty** [comment]
The contract as is parses. This establishes that `mod` works. 

This is very close to [this prior example.](https://github.com/metastatedev/juvix/blob/develop/experimental/juvix/contract.jvx) The rest of that example does not parse atm which is expected.

File: test/examples/Token.ju [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+sig exec : Token.T -> Transaction.T -> Either.T Error Token.T 
+let exec token tx = 
+  case tx.data of 
+  | Transfer _ -> 
+    if | Validation.transfer token tx = Right (transfer token tx) 
+        | else                         = Left NotEnoughFunds 
+  | Mint _ -> 
+    if | Validation.mint token tx = Right (mint token tx) 
+        | else                     = Left NotEnoughFunds
```
Comment (cyberglot) [inline test/examples/Token.ju:122]: slightly misaligned 😭

Comment (thealmarty) [reply test/examples/Token.ju]: Oh right. Fixed!

File: test/examples/Token.ju [hunk 2]
```diff
@@ -0,0 +1,122 @@
+mod Token =
+  let Address = s : String.T {String.length s == 36} 
+
+  type Storage = { 
+    total-supply : Nat.T, 
+    accounts     : Accounts.T { Accounts.measure-value == total-supply } 
+  }
```
Comment (cwgoes) [inline test/examples/Token.ju:7]:
```suggestion
  }
  
```

File: test/examples/Token.ju [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+      , accounts     = new-acc 
+    } 
+
+mod Validation = 
+  let T = Token.T -> Transaction.T -> Bool 
+
+  let mint token tx = 
+    case tx.data of 
+      | Transaction.Mint -> 
+        token.owner == tx-tx-authorized-account
```
Comment (cwgoes) [inline test/examples/Token.ju:87]:
```suggestion
        token.owner == tx.authorized-account 
```

File: test/examples/Token.ju [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+      | Transaction.Mint -> 
+        token.owner == tx-tx-authorized-account 
+      | _ -> 
+        false 
+
+  let transfer token tx = 
+    case tx.data of 
+      | Transaction.Transfer {from-account, amount} -> 
+        has-n token.storage.accounts from-account amount 
+        && tx.authroized-account == from-account
```
Comment (cwgoes) [inline test/examples/Token.ju:95]:
```suggestion
        && tx.authorized-account == from-account 
```

File: test/examples/Token.ju [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        has-n token.storage.accounts from-account amount 
+        && tx.authroized-account == from-account 
+      | _ -> 
+        false 
+
+  let Burn token tx = 
+    case tx.data of 
+      | Transaction.Burn {burn-from-account, burn-ammount} -> 
+        has-n token.storage.accounts burn-from-account burn-amount 
+        && tx.authroized-account == burn-from-account
```
Comment (cwgoes) [inline test/examples/Token.ju:103]:
```suggestion
        && tx.authorized-account == burn-from-account 
```

File: test/examples/Token.ju [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    case tx.data of 
+      | Transaction.Transfer {from-account, amount} -> 
+        has-n token.storage.accounts from-account amount 
+        && tx.authroized-account == from-account 
+      | _ -> 
+        false 
+
+  let Burn token tx = 
+    case tx.data of 
+      | Transaction.Burn {burn-from-account, burn-ammount} ->
```
Comment (cwgoes) [inline test/examples/Token.ju:101]:
```suggestion
      | Transaction.Burn {burn-from-account, burn-amount} -> 
```

File: test/examples/Token.ju [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+  | NotOwnerToken  
+  | NotEnoughTokens 
+
+sig exec : Token.T -> Transaction.T -> Either.T Error Token.T 
+let exec token tx = 
+  case tx.data of 
+    | Transfer _ -> 
+      if | Validation.transfer token tx = Right (transfer token tx) 
+         | else                         = Left NotEnoughFunds 
+  | Mint _ ->
```
Comment (cwgoes) [inline test/examples/Token.ju:120]:
```suggestion
   | Mint _ -> 
```

File: test/examples/Token.ju [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+  | NotOwnerToken  
+  | NotEnoughTokens 
+
+sig exec : Token.T -> Transaction.T -> Either.T Error Token.T 
+let exec token tx = 
+  case tx.data of 
+    | Transfer _ -> 
+      if | Validation.transfer token tx = Right (transfer token tx) 
+         | else                         = Left NotEnoughFunds 
+   | Mint _ ->
```
Comment (cwgoes) [inline test/examples/Token.ju:121]:
```suggestion
    | Mint _ -> 
```

**Comment by cwgoes** [comment]
Tested locally, works fine.

**Pull request merged**